vnext/Microsoft.ReactNative.SharedManaged/ReactContextExtensions.cs

```diff
@@ -56,9 +56,90 @@ public static void CallJSFunction<T1>(this IReactContext reactContext, string mo
     }
 
-    public static void EmitJSEvent<T>(this IReactContext reactContext, string eventEmitterName, string eventName, T arg)
+    public static void EmitJSEvent(this IReactContext reactContext, string eventEmitterName, string eventName)
     {
-      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) => writer.WriteValue(arg));
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) => {});
+    }
+
+    public static void EmitJSEvent<T1>(this IReactContext reactContext, string eventEmitterName, string eventName, T1 arg1)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2>(this IReactContext reactContext, string eventEmitterName, string eventName, T1 arg1, T2 arg2)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+        writer.WriteValue(arg2);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2, T3>(this IReactContext reactContext, string eventEmitterName, string eventName, T1 arg1, T2 arg2, T3 arg3)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+        writer.WriteValue(arg2);
+        writer.WriteValue(arg3);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2, T3, T4>(this IReactContext reactContext,
+      string eventEmitterName, string eventName, T1 arg1, T2 arg2, T3 arg3, T4 arg4)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+        writer.WriteValue(arg2);
+        writer.WriteValue(arg3);
+        writer.WriteValue(arg4);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2, T3, T4, T5>(this IReactContext reactContext,
+      string eventEmitterName, string eventName, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+        writer.WriteValue(arg2);
+        writer.WriteValue(arg3);
+        writer.WriteValue(arg4);
+        writer.WriteValue(arg5);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2, T3, T4, T5, T6>(this IReactContext reactContext,
+      string eventEmitterName, string eventName, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6)
+    {
+      reactContext.EmitJSEvent(eventEmitterName, eventName, (IJSValueWriter writer) =>
+      {
+        writer.WriteValue(arg1);
+        writer.WriteValue(arg2);
+        writer.WriteValue(arg3);
+        writer.WriteValue(arg4);
+        writer.WriteValue(arg5);
+        writer.WriteValue(arg6);
+      });
+    }
+
+    public static void EmitJSEvent<T1, T2, T3, T4, T5, T6, T7>(this IReactContext reactContext,
```

Comment:
>T7 [](start = 59, length = 2)

Does that mean that we are only supporting functions which have up to 6 args?

Reply:
Yes, this is the current limit. C# does not support variadic arguments like C++ and we have to provide a new function for each set of arguments.
I think seven arguments is a reasonable limit to start with. We can expand it should someone have a need for more arguments.
Please note that in many cases developers use a single argument of an object or array type and pass as many values as they want.